src/srb2kart/serverinfo/serverinfo.go

```diff
@@ -0,0 +1,297 @@
+	default:
+		return nil, fmt.Errorf("unknown packet type: %d", header.PacketType)
+	}
+}
+
+func parseFileNeeded(data [915]byte, fileNeededCount int) []FileNeededEntry {
+	var entries []FileNeededEntry
+	buf := bytes.NewBuffer(data[:])
+	for i := 0; i < fileNeededCount; i++ {
+		entry := FileNeededEntry{}
+		binary.Read(buf, binary.LittleEndian, &entry.WillSend)
+		b, _ := buf.ReadByte()
+		entry.WillSend = (b >> 4) == 1
+		binary.Read(buf, binary.LittleEndian, &entry.TotalSize)
+		entry.FileName, _ = buf.ReadString(0)
+		// Trim off the null terminator
+		entry.FileName = entry.FileName[:len(entry.FileName)-1]
+		binary.Read(buf, binary.LittleEndian, &entry.MD5)
+		entries = append(entries, entry)
+	}
+	return entries
+}
+
+func unpackServerInfoPacket(data []byte) (*KartServerInfoPacket, error) {
+	var packetRaw kartServerInfoPacketRaw
+	if err := binary.Read(bytes.NewReader(data), binary.LittleEndian, &packetRaw); err != nil {
+		return nil, err
+	}
+	packet := KartServerInfoPacket{
+		Header:         packetRaw.Header,
+		PacketVersion:  packetRaw.PacketVersion,
+		Application:    packetRaw.Application,
+		Version:        packetRaw.Version,
+		SubVersion:     packetRaw.SubVersion,
+		NumberOfPlayer: packetRaw.NumberOfPlayer,
+		MaxPlayers:     packetRaw.MaxPlayers,
+		GameType:       packetRaw.GameType,
+		ModifiedGame:   packetRaw.ModifiedGame,
+		CheatsEnabled:  packetRaw.CheatsEnabled,
+		KartVars:       packetRaw.KartVars,
+		FileNeededNum:  packetRaw.FileNeededNum,
+		Time:           packetRaw.Time,
+		LevelTime:      packetRaw.LevelTime,
+		ServerName:     nullTerminated(packetRaw.ServerName[:]),
+		MapName:        nullTerminated(packetRaw.MapName[:]),
+		MapTitle:       nullTerminated(packetRaw.MapTitle[:]),
+		MapMD5:         packetRaw.MapMD5,
+		ActNum:         packetRaw.ActNum,
+		IsZone:         packetRaw.IsZone,
+		HttpSource:     nullTerminated(packetRaw.HttpSource[:]),
+		FileNeeded:     parseFileNeeded(packetRaw.FileNeeded, int(packetRaw.FileNeededNum)),
+	}
+	return &packet, nil
+}
+
+func unpackPlayerInfoPacket(data []byte) (*KartPlayerInfoPacket, error) {
+	var packetRaw kartPlayerInfoPacketRaw
+	if err := binary.Read(bytes.NewReader(data), binary.LittleEndian, &packetRaw); err != nil {
+		return nil, err
+	}
+	packet := KartPlayerInfoPacket{
+		Header: packetRaw.Header,
+	}
+	for i := 0; i < len(packetRaw.PlayerInfo); i++ {
+		entry := packetRaw.PlayerInfo[i]
+		packet.PlayerInfo = append(packet.PlayerInfo, KartPlayerInfoEntry{
+			Node: entry.Node,
+			Name: nullTerminated(entry.Name[:]),
+			Address: [4]uint8{
+				entry.Address[0],
+				entry.Address[1],
+				entry.Address[2],
+				entry.Address[3],
+			},
+			Team:         entry.Team,
+			Skin:         entry.Skin,
+			Data:         entry.Data,
+			Score:        entry.Score,
+			TimeInServer: entry.TimeInServer,
+		})
+	}
+	return &packet, nil
+}
+
+func GetSRB2Info(adress string) (*KartServerInfoPacket, *KartPlayerInfoPacket, error) {

```

Comment:
This should be the only public function. 
But also, why pointers. If I can avoid those I will.